README.md

```diff
@@ -20,3 +20,14 @@ randomly without memory instead of systematically trying them?
 
 Describe your reasoning and the conclusion you've come to. Your reasoning is the
 most important part. Add your answer to this markdown file.
+
+The runtime complexity is going to be $O(n!*n)$ because the permute function makes all possible permutations in the array. Each time permute is called there is a swap of two elements and recurses. So this is the $n!$ part. The isSorted is $O(n)$ because it just scans the array.
+So the best case is going to be linear $O(n)$ for n being the size of the array. The permutationSort function uses the isSorted function and the isSorted function returns true with permutationSort returning 0. This is because there is no need to generate permutations because the array is already sorted. The worst case is going to be $O(n!)$ if the array is not sorted abd the algorithm must make every permutation of the array and check if they're sorted.
```

Comment:
What do you have to do for each permutation?

Reply:
And what effect does that have on the worst case complexity?

Reply:
Well if it's not sorted the algorithm will make all the permutations in the array and for each of those permutations it calls isSorted. So, the permutations will take n! because of the swaps from index $i$ and $r$ at each level and there's a level of recursion for each element. The $l+1$ is what's moving the recursion to the next index. Then the isSorted function that's called after each permutation take constant time to check if the array is in order. So for worse case I'm getting $n!$ for each permutation and then the constant time $n$. I forgot to ignore the constant so worse case complexity would be $O(n!)$

Reply:
Oh because it changes depending on the amount of elements are in the input. So then the worst case would be $O(n!*n)$